apps/03TicTacToe.js

```diff
@@ -22,23 +22,37 @@ function printBoard() {
 }
 
 function horizontalWin() {
-    // Your code here
+  if( (board[0][0] === playerTurn && board[0][1] === playerTurn && board[0][2] === playerTurn)||(board[1][0] === playerTurn && board[1][1] === playerTurn && board[1][2] === playerTurn)||(board[2][0] === playerTurn && board[2][1] === playerTurn && board[2][2] === playerTurn)){
+    return true;
+  }
 }
 
 function verticalWin() {
-    // Your code here
+  if( (board[0][0] === playerTurn && board[1][0] === playerTurn && board[2][0] === playerTurn)||(board[0][1] === playerTurn && board[1][1] === playerTurn && board[2][1] === playerTurn)||(board[0][2] === playerTurn && board[1][2] === playerTurn && board[2][2] === playerTurn)){
+ 		return true;
+ 	}
 }
 
 function diagonalWin() {
-    // Your code here
+  if( (board[0][0] === playerTurn && board[1][1] === playerTurn && board[2][2] === playerTurn)||(board[0][2] === playerTurn && board[1][1] === playerTurn && board[2][0] === playerTurn)){
+ 		return true;
+ 	}
 }
 
 function checkForWin() {
-    // Your code here
+  if (horizontalWin()=== true || verticalWin()=== true || diagonalWin()===true){
+    console.log('Player ' + playerTurn + ' Won!');
+    return true;
+  }
 }
 
 function ticTacToe(row, column) {
-    // Your code here
+    board[row][column]= playerTurn;
+//    if (checkForWin === true){
+//      return;
+//    }
```

Comment:
App works as expected! When I do node 03TicTacToe.js though, i noticed it is not giving me a message when Player X or O wins. It may have to do with this commented segment here. We would just need to do this:

```javascript
board[row][column] = playerTurn;
checkForWin();
....
```